src/py_pecg/Subfunction_Module.py

```diff
@@ -0,0 +1,332 @@
+import numpy as np
+
+
+def buscamin(x: np.ndarray) -> int:
+    """
+    This function searches the first local minimum of the modulus of an array x,
+    While truncating the edges (First and Final Indices).
+
+    Args:
+        x (np.ndarray): The target array.
+
+    Returns:
+        ind (int): The first index in x for which the first minimum is located.
+
+    Credits:
+        Original authors: Juan Pablo Martínez, Rute Almeida, Salvador Olmos,
+        Ana Paula Rocha, Pablo Laguna
+        Original Publication: A wavelet-based ECG delineator:
+        evaluation on standard databases
+        Original DOI: 10.1109/TBME.2003.821031.
+    """
+    x = np.abs(x)
+    localmin = (x[1:-1] <= x[:-2]) & (x[1:-1] <= x[2:])
+    ind = np.argmax(localmin)
+    return ind
+
+
+def searchon(piconset: int, sig: np.ndarray, K: float) -> int:
+    """
+    This function searches the onset of a wave using the derivative method.
+
+    Args:
+        piconset (int): Position of the first relevant onset in the wavelet.
+        sig (np.ndarray): Wavelet Signal (in a single scale).
+        K (float): Threshold Factor.
+
+    Returns:
+        onset (int): Resultant index of the wave onset.
+
+    Credits:
+        Original authors: Juan Pablo Martínez, Rute Almeida, Salvador Olmos,
+        Ana Paula Rocha, Pablo Laguna
+        Original Publication: A wavelet-based ECG delineator:
+        evaluation on standard databases
+        Original DOI: 10.1109/TBME.2003.821031.
+    """
+    if K == 0:
+        raise ZeroDivisionError("K cannot be zero!")
+
+    if sig.ndim > 1 and sig.shape[0] > 1 and sig.shape[1] > 1:
+        raise ValueError(
+            "Input is a matrix or higher-dimensional array, not a row or column vector."
+        )
+    elif sig.ndim > 2:
+        raise ValueError("Input is a higher-dimensional array, not a vector.")
+
+    if piconset.size == 0 or sig.size == 0:
+        onset = []
+
+    else:
+        maxderon = abs(sig[-1])
+
+        # Maximum Derivative
+        ind1 = (
+            min(np.argwhere(abs(np.flipud(sig[0:-1])) < maxderon / K))
+            if np.any(abs(np.flipud(sig[0:-2])) < maxderon / K)
+            else np.array([])
+        )
```

Comment:
This is a rather hard to understand line. I'd split and simplify it for readability:
```suggestion
        flipped_signal = np.flipud(sig[:-1])
        condition_indices = np.where(abs(flipped_signal) < maxderon / K)[0]

        if condition_indices.size > 0:
             ind1 = condition_indices[0]
        else:
            ind1 = np.array([])
```

Reply:
I'm not sure why we want `ind1` to be either an integer or an empty NumPy array... as you treat `ind1=np.array([])` and `ind1=0` in the same way on the next lines, you can just set it to 0 if the `condition_indices` is empty and then check `if not ind1` instead of `if ind1.size == 0 or ind1 == 0`